__tests__/app.test.js

```diff
@@ -1,46 +1,83 @@
+/** * @jest-environment node */
 const request = require('supertest');
-const app = require('../src/app');
 const nock = require('nock');
+const app = require('../src/app');
+const { mockListOfJokes, mockRandomJokes, mockPersonalJokes } = require('../data/test-data');
 
-describe('My jokes API', () => {
-  describe('API GET /', () => {
-    it('should respond with a welcome message', done => {
-      request(app)
-        .get('/')
-        .then(res => {
-          expect(res.statusCode).toEqual(200);
-          expect(res.body.message).toEqual('Welcome to my jokes API!');
-          done();
-        });
-    });
-  });
-  it('that this is the jokes endpoint', done => {
+describe('GET / - Homepage', () => {
+  it('should respond with some homepage markup', async () => {
     request(app)
-      .get('/jokes')
+      .get('/')
       .then(res => {
         expect(res.statusCode).toEqual(200);
-        expect(res.body.jokes).toEqual(
-          'You have reached the jokes endpoint, please laugh with care',
-        );
-        done();
+        expect(res.text).toContain('Hello, Welcome to my jokes API');
       });
   });
-  it('should respond with a random joke message', done => {
-    request(app)
+});
+
+describe('GET /jokes', () => {
+  it('that jokes endpoint responds with list of jokes', async () => {
+    nock('https://api.icndb.com')
```

Comment:
As you use nock with the same endpoint multiple times, you could save this to a constant at the top of your file and refer to it each time. What you have done is correct though, just another way of doing it :)